### ISSUE TITLE
TPC-H 2 segfaults with scheduler (fixed in #1625)

### ISSUE THREAD

**Issue by mrks** [hyrise/hyrise#1623]
```
$ cat ../bisect.sh
git submodule update
cd build
ninja hyriseBenchmarkTPCH
./hyriseBenchmarkTPCH --scheduler -q 2 || return 1
cd ..
$ git bisect start master eecc9bd4bc30f8bf507e4e86d6f73613f4ad3d62
$ git bisect run ../bisect.sh
```

```
f0d316cb0e363efd2cd53486eaf2ea6ee865641b is the first bad commit
commit f0d316cb0e363efd2cd53486eaf2ea6ee865641b
Date:   Mon Apr 8 16:03:38 2019 +0200

    [DYOD18/19] Subquery to Join Reformulation (#1473)
```

**Comment by mrks** [comment]
Something that could also be covered by #1103.

**Comment by mrks** [comment]
```
Thread 105 "hyriseBenchmark" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffe97dff700 (LWP 81954)]
0x0000000002a7cee2 in opossum::MultipleChunkReferenceSegmentAccessor<int>::access (this=0x7ffe426131c0, offset=1603) at ../src/lib/storage/segment_accessor.hpp:82
82	    return _accessors[chunk_id]->access(row_id.chunk_offset);
(gdb) bt
#0  0x0000000002a7cee2 in opossum::MultipleChunkReferenceSegmentAccessor<int>::access (this=0x7ffe426131c0, offset=1603) at ../src/lib/storage/segment_accessor.hpp:82
#1  0x000000000273cf43 in opossum::MultiPredicateJoinEvaluator::FieldComparator<std::equal_to<void>, int, int>::compare (this=0x7ffe42613180, left=..., right=...) at ../src/lib/operators/multi_predicate_join/multi_predicate_join_evaluator.hpp:41
#2  0x000000000272f80e in opossum::MultiPredicateJoinEvaluator::satisfies_all_predicates (this=0x7ffe441fc118, left_row_id=..., right_row_id=...) at ../src/lib/operators/multi_predicate_join/multi_predicate_join_evaluator.cpp:51
#3  0x000000000259e1c1 in void opossum::probe_semi_anti<float, float, false>(opossum::RadixContainer<float> const&, std::vector<std::optional<ska::bytell_hash_map<float, boost::container::small_vector<opossum::RowID, 1ul, boost::container::new_allocator<opossum::RowID> >, std::hash<float>, std::equal_to<float>, std::allocator<std::pair<float, boost::container::small_vector<opossum::RowID, 1ul, boost::container::new_allocator<opossum::RowID> > > > > >, std::allocator<std::optional<ska::bytell_hash_map<float, boost::container::small_vector<opossum::RowID, 1ul, boost::container::new_allocator<opossum::RowID> >, std::hash<float>, std::equal_to<float>, std::allocator<std::pair<float, boost::container::small_vector<opossum::RowID, 1ul, boost::container::new_allocator<opossum::RowID> > > > > > > > const&, std::vector<opossum::PosList, std::allocator<opossum::PosList> >&, opossum::JoinMode, opossum::Table const&, opossum::Table const&, std::vector<opossum::OperatorJoinPredicate, std::allocator<opossum::OperatorJoinPredicate> > const&)::{lambda()#1}::operator()() const (
    this=0x7ffe4dc09a40) at ../src/lib/operators/join_hash/join_hash_steps.hpp:557
#4  0x000000000259dded in std::_Function_handler<void (), void opossum::probe_semi_anti<float, float, false>(opossum::RadixContainer<float> const&, std::vector<std::optional<ska::bytell_hash_map<float, boost::container::small_vector<opossum::RowID, 1ul, boost::container::new_allocator<opossum::RowID> >, std::hash<float>, std::equal_to<float>, std::allocator<std::pair<float, boost::container::small_vector<opossum::RowID, 1ul, boost::container::new_allocator<opossum::RowID> > > > > >, std::allocator<std::optional<ska::bytell_hash_map<float, boost::container::small_vector<opossum::RowID, 1ul, boost::container::new_allocator<opossum::RowID> >, std::hash<float>, std::equal_to<float>, std::allocator<std::pair<float, boost::container::small_vector<opossum::RowID, 1ul, boost::container::new_allocator<opossum::RowID> > > > > > > > const&, std::vector<opossum::PosList, std::allocator<opossum::PosList> >&, opossum::JoinMode, opossum::Table const&, opossum::Table const&, std::vector<opossum::OperatorJoinPredicate, std::allocator<opossum::OperatorJoinPredicate> > const&)::{lambda()#1}>::_M_invoke(std::_Any_data const&) (__functor=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/std_function.h:297
```

**Comment by mrzzzrm** [comment]
There is a small chance I already fixed this in #1577 - will check today

**Comment by mrzzzrm** [comment]
Nope, also segfaults :(
